### ISSUE TITLE
Interpret user-provided regexes in rule options with the `u` flag

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/eslint#11423]
**What rule do you want to change?**

`camelcase`, `capitalized-comments`, `default-case`, `dot-notation`, `handle-callback-err`, `id-match`, `line-comment-position`, `lines-around-comment`, `max-len`, `new-cap`, `no-fallthrough`, `no-unused-vars`, `spaced-comment` and `valid-jsdoc`. (These are all of the rules that have options which are interpreted as regular expressions.)

**Does this change cause the rule to produce more or fewer warnings?**

In the vast majority of cases, the behavior would be the same. In cases where the source text contains astral symbols or other non-BMP characters, the rule may produce more or fewer warnings.

**How will the change be implemented? (New option, new default behavior, etc.)?**

This will be a new default behavior. I think this is the behavior that most users expect anyway, although they might not have noticed that the current behavior is different from their expectation.

[This page](https://mathiasbynens.be/notes/es6-unicode-regex) contains a good summary of unicode regexes.

**Please provide some example code that this change will affect:**

```js
/* eslint max-len: [
  error,
  25,
  {ignorePattern: "𝌆{2}"}
] */

var longNameLongNameLongName = '𝌆𝌆';
```

**What does the rule currently do for this code?**

The rule reports an error on the line with a variable declaration, even though it seems like that line should match the `ignorePattern`.

**What will the rule do after it's changed?**

The rule will no longer report an error, because it will interpret the ignore pattern as a unicode regex.

**Are you willing to submit a pull request to implement this change?**

Yes

**Comment by not-an-aardvark** [comment]
This issue was accepted in [today's TSC meeting](https://github.com/eslint/tsc-meetings/blob/3e40976a2ebb9481405d4c3a358bc3191ecd234a/notes/2019/2019-02-28.md#interpret-user-provided-regexes-in-rule-options-with-the-u-flag).
